_ingest-pipelines/processors/text-chunking.md

```diff
@@ -0,0 +1,315 @@
+{: .note}
+
+### Delimiter algorithm
+
+The following table lists the optional parameters for the `delimiter` algorithm.
+
+| Parameter  | Data type | Required/Optional  | Description  |
+|:---|:---|:---|:---|
+| `delimiter`	| String	    | Optional	 | A string delimiter used to split text. You can set the `delimiter` to any string, for example, `\n` (split text into paragraphs on new line) or `.` (split text into sentences). Default is `\n\n` (split the text into paragraphs on two new line characters). |
+| `max_chunk_limit`	 | Integer	   | Optional	 | The chunk limit for chunking algorithms. Default is `100`. To disable this parameter, set it to `-1`.	 |
+
+The `max_chunk_limit` parameter restricts the number of chunked passages. If the number of passages generated by the processor exceed this limit, the algorithm will return an exception, prompting you to either increase or disable this limit.
+{: .note}
+
+## Using the processor
+
+Follow these steps to use the processor in a pipeline. You can specify the chunking algorithm when creating the processor. If you don't provide an algorithm name, the chunking processor will use default `fixed_token_length` algorithm along with all its default parameters.
+
+**Step 1: Create a pipeline.**
+
+The following example request creates an ingest pipeline that converts the text in the `passage_text` field into chunked passages, which will be stored in the `passage_chunk` field:
+
+```json
+PUT _ingest/pipeline/text-chunking-ingest-pipeline
+{
+  "description": "A text chunking ingest pipeline",
+  "processors": [
+    {
+      "text_chunking": {
+        "algorithm": {
+          "fixed_token_length": {
+            "token_limit": 10,
+            "overlap_rate": 0.2,
+            "tokenizer": "standard"
+          }
+        },
+        "field_map": {
+          "passage_text": "passage_chunk"
+        }
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+**Step 2 (Optional): Test the pipeline.**
+
+It is recommended that you test your pipeline before you ingest documents.
+{: .tip}
+
+To test the pipeline, run the following query:
+
+```json
+POST _ingest/pipeline/text-chunking-ingest-pipeline/_simulate
+{
+  "docs": [
+    {
+      "_index": "testindex",
+      "_id": "1",
+      "_source":{
+         "passage_text": "This is an example document to be chunked. The document contains a single paragraph, two sentences and 24 tokens by standard tokenizer in OpenSearch."
+      }
+    }
+  ]
+}
+```
+{% include copy-curl.html %}
+
+#### Response
+
+The response confirms that in addition to the `passage_text` field, the processor has generated chunking results in the `passage_chunk` field. The processor split the paragraph into 10-word chunks. Because of the `overlap` setting of 0.2, the last two words of a chunk are duplicated in the following chunk:
```

Comment:
```suggestion
The response confirms that, in addition to the `passage_text` field, the processor has generated chunking results in the `passage_chunk` field. The processor split the paragraph into 10-word chunks. Because of the `overlap` setting of 0.2, the last 2 words of a chunk are duplicated in the following chunk:
```